### PR TITLE
apply selectors before checking for requirements

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: conda_build/variants.py [hunk 1]
```diff
@@ -555,7 +555,10 @@ def find_used_variables_in_text(variant, recipe_text):
         conditional_regex = r"(?:^|[^\{])\{%\s*(?:el)?if\s*" + v_regex + r"\s*(?:[^%]*?)?%\}"
         # plain req name, no version spec.  Look for end of line after name, or comment or selector
         requirement_regex = r"^\s+\-\s+%s\s*(?:\s[\[#]|$)" % v_req_regex
-        all_res.extend([variant_regex, selector_regex, conditional_regex, requirement_regex])
+        if not selectors:
```
Comment (msarahan) [inline conda_build/variants.py]: This logic is counterintuitive to me.  I think it means:

**Comment by msarahan** [comment]
Aside from my one question, this looks fine.  I'll be happy to merge if you can clarify things a bit for me.

File: conda_build/variants.py (hunk 1 above)
Comment (isuruf) [reply conda_build/variants.py]: It should have been `if selectors`.

Comment (msarahan) [reply conda_build/variants.py]: Could you please document how you want this to behave?  I still don't trust myself to get this right, especially without some tests.

File: conda_build/metadata.py [hunk 2]
```diff
@@ -2063,31 +2063,47 @@ def get_used_vars(self, force_top_level=False, force_global=False):
...
         else:
             if is_output and not force_top_level:
-                recipe_text = self.extract_single_output_text(self.name(), apply_selectors=False)
+                recipe_text = self.extract_single_output_text(self.name(),
+                                apply_selectors=apply_selectors)
             else:
                 recipe_text = (self.get_recipe_text(force_top_level=force_top_level,
-                                                    apply_selectors=False).replace(
+                                                    apply_selectors=apply_selectors).replace(
                                     self.extract_outputs_text(apply_selectors=False).strip(), '') +
```
Comment (isuruf) [inline conda_build/metadata.py]: This should have been `apply_selectors=apply_selectors`.

cc @CJ-Wright

File: conda_build/metadata.py [hunk 3]
```diff
@@ -2063,31 +2063,47 @@ def get_used_vars(self, force_top_level=False, force_global=False):
...
                             self.extract_single_output_text(self.name(), apply_selectors=False))
```
Comment (isuruf) [inline conda_build/metadata.py]: Same here

**Pull request merged**